components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -696,14 +696,34 @@ private List<Integer> getUserConsentClaimIds(OAuthMessage oAuthMessage) {
 
     private void addSessionDataKeyToSessionDataCacheEntry(OAuthMessage oAuthMessage) {
 
-        Cookie cookie = FrameworkUtils.getAuthCookie(oAuthMessage.getRequest());
-        if (cookie != null) {
-            String sessionContextKey = DigestUtils.sha256Hex(cookie.getValue());
+        String commonAuthIdCookieValue = getCommonAuthCookieString(oAuthMessage.getRequest());
+        if (StringUtils.isNotBlank(commonAuthIdCookieValue)) {
+            String sessionContextKey = DigestUtils.sha256Hex(commonAuthIdCookieValue);
             oAuthMessage.getSessionDataCacheEntry().getParamMap().put(FrameworkConstants.SESSION_DATA_KEY, new String[]
                     {sessionContextKey});
         }
     }
 
+    /**
+     * Retrieves the value of the commonAuthId cookie either from request cookies if available or
+     * from the request attribute where the response header contains commonAuthId value.
+     *
+     * @param request HttpServletRequest An authorization or authentication request.
+     * @return String commonAuthId value.
```

Comment:
Shall we mention that there is a chance of returning null? WDYT?

Reply:
Fixed ad3cb637ce2bd2485bc1b335e6f68be0d87ef70b